src/services/FulfillmentService.js

```diff
@@ -49,6 +55,46 @@ class FulfillmentService {
       throw err;
     }
   };
+
+  closeTheLoop = async (event, context, callback) => {
+    context.callbackWaitsForEmptyEventLoop = false;
+
+    try {
+      const now = new Date();
+      const today = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+      const followUps = await FollowUp.find({ delivery_date: today }).exec();
+      followUps.forEach(followUp => {
+        const emailLog = new EmailMessageLog({
+          initialRequestType: 'CLOSE-THE-LOOP',
+          initialRequestDate: new Date(),
+          initialRequest: JSON.parse(event.Records[0].body),
+          generateEmailMessageDate: null,
+          generateEmailMessage: null,
+          deliverEmailMessageDate: null,
+          deliverEmailMessage: null,
+        });
+        emailLog.save();
+        const params = {
+          MessageBody: JSON.parse(event.Records[0].body).message,
```

Comment:
The message body here should be the information from the original message for the email that this followup is for. So we need to change the followup.data field's definition to accept any object instead of a string like it is currently doing. Then, we need to update the EmailService.sendEmail method so that it stores the original message's body in the followup.data field. Lastly, we should update this line so that it sets the message body to the followup.data field.

Reply:
as I understand at this moment we don't know what kind of object we get in sendEmail func. maybe better do it when we will be known it